packages/mobx-state-tree/src/core/mst-operations.ts

```diff
@@ -451,6 +451,47 @@ export function resolveIdentifier(
     return node ? node.value : undefined
 }
 
+/**
+ * Returns the identifier of the target node. If a property is supplied returns the identifier
+ * of the property even if it is a reference to a missing object.
+ *
+ * @export
+ * @param {IStateTreeNode} target
+ * @param {string} property
+ * @returns {(string | null)}
+ */
+export function getIdentifier<T extends IStateTreeNode>(
+    target: T,
+    property?: keyof T
```

Comment:
not so sure the property param is needed - omitting it could clean up the code and would be less error prone since string literals can't be checked on e.g. renaming a property.

was there a specific use case you had in mind?

Reply:
The property param is for the use case in my original issue #641. 

The idea is that you can access the identifier that is being searched for by a reference, regardless of whether that identifier exists in the tree or not.

I want to be able to do this so I can lazily load references. This can be done using Custom References, but loading them on first access is a bit unpredictable, which I don't like. I know which references are required for each view, so I'd rather load them all at the start in one place. Currently you have to snapshot each object in order to find out the identifiers that need loading - which is not the intended usage of the snapshot feature.

I've added some tests in `packages/mobx-state-tree/test/node.ts` which cover these cases.